R/compare_diagnoses.R

```diff
@@ -0,0 +1,331 @@
+#'
+#' @param base_design A design or a diagnosis.
+#' @param comparison_design A design or a diagnosis.
+#' @param sims The number of simulations, defaulting to 1000. sims may also be a vector indicating the number of simulations for each step in a design, as described for \code{\link{simulate_design}}. Used for both designs.
+#' @param bootstrap_sims_base Number of bootstrap replicates for the diagnosands to obtain the standard errors of the diagnosands, defaulting to \code{1000}. Set to FALSE to turn off bootstrapping. Used for base_design.
+#' @param bootstrap_sims_comparison Number of bootstrap replicates for the diagnosands to obtain the standard errors of the diagnosands, defaulting to \code{1000}. Set to FALSE to turn off bootstrapping. Used for comparison_designs.
+#' @param merge_by_estimator A logical. Whether to include \code{estimator_label} in the set of columns used for merging. Defaults to \code{TRUE}
+#' @param alpha The confidence level, 0.01 by default.
+#'  @return A list with a data frame of compared diagnoses and both diagnoses.
+#'
+#' @details
+#' 
+#' The function \code{compare_diagnoses()} runs a many-to-many merge matching by \code{estimand_label} and \code{term} (if present). If  \code{merge_by_estimator} equals \code{TRUE}, \code{estimator_label} is also included in the merging condition. Any diagnosand that is not included in both designs will be dropped from the merge.
+#' 
+#' The data frame of compared diagnoses has a column, \code{in_interval}, that indicates if two given diagnosands diverge statistically. 
+#' i.e. if a diagnosand from \code{comparison_design} is not contained in the 95\% bootstrap confidence interval of their equivalent diagnosand from \code{base_design}. The column \code{in_interval} equals zero for divergent diangnosands and one otherwise..
+#'
+#' @examples
+#' design_a <- declare_population(N = 100, u = rnorm(N), X = runif(N, 0, 2)) +
+
+#' declare_potential_outcomes(
+#'   Y_Z_0 = u, 
+#'   Y_Z_1 = u + rnorm(N, mean = 2, sd = 2)) +
+#' declare_assignment() + 
+#'
+#' declare_estimand(ATE = mean(Y_Z_1 - Y_Z_0), label = "ATE") +
+#' 
+#' declare_reveal() +
+#' 
+#' declare_estimator(Y ~ Z, estimand = "ATE", label = "est1")
+#' 
+#' design_b <- replace_step(design_a, step = "assignment", declare_assignment(prob = 0.3) )
+#' 
+#' compare_diagnoses(design_a, design_b)
+#'  
+#' @export
+compare_diagnoses <- function(base_design,
+                              comparison_design, 
+                              sims = 1000,
+                              bootstrap_sims_base = 1000, 
+                              bootstrap_sims_comparison = 100,
+                              alpha = 0.01,
+                              merge_by_estimator = TRUE){
+  
+  
+  if(bootstrap_sims_base <= 99) stop("Please choose a higher number of bootstrap simulations")
+  
+  
+ # Diagnose designs if base/comparison are of class "design"
+  
+  if( "design" %in% class(base_design)  ){
```

Comment:
Please use check_design_class_single(base_design) here

Reply:
Ignoring this suggestion since function should also work for objects of classe `diagnosis`

Reply:
Also you can use the `inherits` function instead.